server.js

```diff
@@ -44,12 +45,8 @@ var renderHtml = function(url, cb) {
     }; //For service purposes only
 
     page.onInitialized = function() {
-       page.evaluate(function() {
-            setTimeout(function() {
-                console.debug("setTimeout() rang, calling callPhantom()");
-                window.callPhantom();
-            }, 10000); //If no callPhantom script was called on the page, then wait 10secs and load page anyway.
-        });
+       watch.angular(page); //Remove that if you're not using angular
+       watch.timeout(page);
```

Comment:
At first glance I thought the timeout thing, watches if there are any timeouts running on the page, but it isn't.